File: 88.merge-sorted-array.py
#
# @lc app=leetcode id=88 lang=python3
#
# [88] Merge Sorted Array
#

# @lc code=start
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        for i in range(0,n):
            nums1.pop()
        for i, n1 in enumerate(nums1):
            if nums2 and nums2[0] <= n1: nums1.insert(i, nums2.pop(0))

        # remainders on nums2
        nums1.extend(nums2)
        
        return nums1
        
# @lc code=end

File: 88.merge-sorted-array.js
/*
 * @lc app=leetcode id=88 lang=javascript
 *
 * [88] Merge Sorted Array
 */

// @lc code=start
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
  let nums3 = [...nums1.slice(0,m), ...nums2].sort((a,b)=>a-b)
  for (let i=0;i<nums1.length;i++) {
    nums1[i] = nums3[i]
  }
};
// @lc code=end

